File: 005/divisible.go
package main

import "fmt"

func GCD(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func main() {
	lcm := 1
	for i := 2; i <= 20; i++ {
		lcm *= i / GCD(i, lcm)
	}
	fmt.Println(lcm)
}

File: 005/divisible.rs
fn gcd(mut a: u64, mut b: u64) -> u64 {
    while b != 0 {
        let t = a;
        a = b;
        b = t % b;
    }
    a
}

fn main() {
    let lcm = (2..21).fold(1, |acc, i| {
        acc * i / gcd(i, acc)
    });
    println!("{}", lcm);
}
